app/routes/user/email_notifications.server.ts

```diff
@@ -0,0 +1,216 @@
+    const item = (await db.email_notification.get({
+      sub: this.#sub,
+      uuid,
+    })) as ({ sub: string } & EmailNotificationVM) | null
+    if (!item) throw new Response(null, { status: 404 })
+    item.noticeTypes = item.topics.map((topic) => {
+      return topicToFormatAndNoticeType(topic).noticeType
+    })
+    item.format = topicToFormatAndNoticeType(item.topics[0]).noticeFormat
+    const { sub, ...notification } = item
+    return {
+      uuid,
+      ...notification,
+    }
+  }
+
+  // Update
+  async updateEmailNotification(email_notification: EmailNotification) {
+    if (!email_notification.uuid) return null
+    const db = await tables()
+    await db.email_notification.update({
+      Key: { sub: this.#sub, uuid: email_notification.uuid },
+      UpdateExpression:
+        'set #name = :name, #recipient = :recipient, #topics = :topics, #active = :active ',
+      ExpressionAttributeNames: {
+        '#name': 'name',
+        '#topics': 'topics',
+        '#recipient': 'recipient',
+        '#active': 'active',
+      },
+      ExpressionAttributeValues: {
+        ':name': email_notification.name,
+        ':recipient': email_notification.recipient,
+        ':topics': email_notification.topics,
+        ':active': email_notification.active,
+      },
+    })
+
+    // Update Materialized View
+    const subscriptions = await db.email_notification_subscription.query({
+      KeyConditionExpression: '#uuid = :uuid',
+      ExpressionAttributeNames: {
+        '#uuid': 'uuid',
+      },
+      ExpressionAttributeValues: {
+        ':uuid': email_notification.uuid,
+      },
+    })
+    await Promise.all(
+      subscriptions.Items.map((sub) => {
+        return db.email_notification_subscription.delete({
+          uuid: sub.uuid,
+          topic: sub.topic,
+        })
+      })
+    )
+    if (email_notification.active) {
+      await Promise.all(
+        email_notification.topics.map((topic) => {
+          return db.email_notification_subscription.put({
+            uuid: email_notification.uuid,
+            topic,
+            recipient: email_notification.recipient,
+          })
+        })
+      )
+    }
+  }
+
+  // Delete
+  async deleteEmailNotification(uuid: string) {
+    const db = await tables()
+    const item = await db.email_notification.get({
+      sub: this.#sub,
+      uuid,
+    })
+    if (!item) throw new Response(null, { status: 404 })
+    await db.email_notification.delete({ sub: this.#sub, uuid })
+    const subscriptions = await db.email_notification_subscription.query({
+      KeyConditionExpression: '#uuid = :uuid',
+      ExpressionAttributeNames: {
+        '#uuid': 'uuid',
+      },
+      ExpressionAttributeValues: {
+        ':uuid': uuid,
+      },
+    })
+    await Promise.all(
+      subscriptions.Items.map((sub) => {
+        return db.email_notification_subscription.delete({
+          uuid: sub.uuid,
+          topic: sub.topic,
+        })
+      })
```

Comment:
```suggestion
      subscriptions.Items.map((sub) => (
        db.email_notification_subscription.delete({
          uuid: sub.uuid,
          topic: sub.topic,
        ))
      })
```

Reply:
fixed in commit, the end parenthesis is mismatched